File: 3162-find-the-number-of-good-pairs-i/3162-find-the-number-of-good-pairs-i.cs
public class Solution {
    public int NumberOfPairs(int[] nums1, int[] nums2, int k) {
        var good = 0;
        foreach(var n2 in nums2)
            foreach(var n1 in nums1)
                if(n1 % (n2 * k) == 0)
                    good++;

        return good;
    }
}

File: 3162-find-the-number-of-good-pairs-i/3162-find-the-number-of-good-pairs-i.java
class Solution {
    public int numberOfPairs(int[] nums1, int[] nums2, int k) {
        var good = 0;
        for(var n2 : nums2)
            for(var n1 : nums1)
                if(n1 % (n2 * k) == 0)
                    good++;

        return good;
    }
}